src/Illuminate/Queue/CallQueuedHandler.php

```diff
@@ -59,6 +59,10 @@ public function call(Job $job, array $data)
             return $this->handleModelNotFound($job, $e);
         }
 
+        if ($command->uniqueUntilStart ?? false) {
+            $this->ensureUniqueJobLockIsReleased($command);
```

Comment:
If we release the lock here and then also release it when the job is finished processing then don't we risk releasing the lock for a job that was queued after this one started processing but before it finishes processing? Then we will release the lock and there could be duplicate jobs on the queue that haven't started processing yet.